src/dmd/traits.d

```diff
@@ -1420,7 +1420,9 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         auto s = getDsymbol(o);
         if (!s)
         {
-            e.error("argument has no members");
+            e.error("In expression `%s` `%s` can't have members", e.toChars(), o.toChars());
+            e.errorSupplemental("`%s` must evaluate to either a module, a struct, an union, a class, an interface or a template (not a template declaration)", o.toChars());
```

Comment:
Maybe refer to template instantiations instead of excluding template declarations?

Reply:
Ok